numba/tests/test_record_dtype.py

```diff
@@ -1495,6 +1513,20 @@ def test_set_arrays(self):
             arr_res = cfunc(nbarr)
             np.testing.assert_equal(arr_res, arr_expected)
 
+    def test_setitem(self):
+        nbarr1 = np.recarray(2, dtype=recordwith2darray)
+        nbarr1[0] = np.array([(1, ((1, 2), (4, 5), (2, 3)))],
+                             dtype=recordwith2darray)[0]
+        nbarr2 = np.recarray(2, dtype=recordwith2darray)
+        nbarr2[0] = np.array([(10, ((10, 20), (40, 50), (20, 30)))],
+                             dtype=recordwith2darray)[0]
+        pyfunc = record_setitem_array
+        args = nbarr1[0], nbarr2[0]
+        arr_expected = pyfunc(*args)
```

Comment:
The function `record_setitem_array` doesn't return anything so the default return of `None` is used. As a result the check at the end of this test is just checking that `None` equals `None`?

Reply:
I think https://github.com/stuartarchibald/numba/commit/6042b7fbf1719de9136ca5acdfaaa7673f371308 fixes, feel free to incorporate.

Reply:
Thanks for the fix, I've just incorporated it.